src/interaction/MeasureLength.js

```diff
@@ -49,15 +73,74 @@ olBase.inherits(exports, ngeoInteractionMeasure);
 /**
  * @inheritDoc
  */
-exports.prototype.createDrawInteraction = function(style, source) {
+exports.prototype.createDrawInteraction = function(style) {
   return new olInteractionDraw({
     type: /** @type {ol.geom.GeometryType} */ ('LineString'),
-    source: source,
+    geometryFunction: (...args) =>
+      this.linestringGeometryFunction(...args),
+    condition: () => true,
     style: style
   });
 };
 
 
+/**
+ * Create a `linestringGeometryFunction` that will create a line string with segments
+ * snapped to π/4 angle.
+ * Use this with the draw interaction and `type: 'LineString'`.
+ * @param {LineCoordType} coordinates Coordinates.
+ * @param {LineString=} opt_geometry Geometry.
+ * @return {LineString} Geometry.
+ */
+exports.prototype.linestringGeometryFunction = function(coordinates, opt_geometry) {
+  if (modifierPressed && this.tolerance !== undefined && this.source !== undefined) {
+    const viewRotation = this.getMap().getView().getRotation();
+    const angle = Math.PI / 4;
+    const from = coordinates[coordinates.length - 2];
+    const to = coordinates[coordinates.length - 1];
+    const dx = from[0] - to[0];
+    const dy = from[1] - to[1];
+    const length = distance(from, to);
+    const rotation = viewRotation + Math.round((Math.atan2(dy, dx) - viewRotation) / angle) * angle;
+
+    to[0] = from[0] - (length * Math.cos(rotation));
+    to[1] = from[1] - (length * Math.sin(rotation));
+
+    const delta = this.getMap().getView().getResolution() * this.tolerance;
+    const bbox = [to[0] - delta, to[1] - delta, to[0] + delta, to[1] + delta];
+
+    const layerSource = this.source;
+    const featuresInExtent = layerSource.getFeaturesInExtent(bbox);
+    if (featuresInExtent.length > 0) {
+      featuresInExtent.forEach((feature) => {
```

Comment:
Here you should introduce:
- a `lastIntersection` array that you pass below as last parameter of `computeLineSegmentIntersection`;
- a `bestIntersection` array that contains the best intersection (= the one that minimizes the distance2 between from and lastIntersection; initialized with `to`).